src/bootchooser.c

```diff
@@ -1159,11 +1159,26 @@ static gboolean efi_bootorder_get(GList **bootorder_entries, GList **all_entries
 	}
 
 	while (g_match_info_matches(match)) {
+		gchar **name_data_splits = NULL;
+		gchar *name_and_data = g_match_info_fetch(match, 2);
+		name_data_splits = g_strsplit(name_and_data, "\t", 0);
+		if (g_strv_length (name_data_splits) == 0) {
+			g_set_error(
+					error,
+					R_BOOTCHOOSER_ERROR,
+					R_BOOTCHOOSER_ERROR_FAILED,
+					"Extracting a boot entry name failed!");
+			res = FALSE;
+			goto out;
+		}
+
 		efi_bootentry *entry = g_new0(efi_bootentry, 1);
 		entry->num = g_match_info_fetch(match, 1);
-		entry->name = g_match_info_fetch(match, 2);
+		entry->name = g_strdup(name_data_splits[0]);
```

Comment:
Instead of using `g_strsplit`, you can also modify entry->name in place. Use strchr to find the first tab, and if there is one, replace it by `'\0'`. This also avoids any additional memory allocations.

Reply:
Ah good point. I'll try that... might take me some time though.
Better than to leak memory.. phew :sweat_smile:

Reply:
I'll try with cleaning up my current code further, as the glib provided functions are easier for me to use right now.
I'd be happy to see whether with the recent change I am circumventing the memory leak already.